src/lib/components/FieldSelect.svelte

```diff
@@ -97,15 +120,37 @@
 		</div>
 
 		<Combobox.Content sideOffset={4} class="field-combobox-content">
-			{#each filteredOptions as option}
-				<Combobox.Item value={option.value} label={option.label} class="field-combobox-item">
-					<Combobox.ItemIndicator class="field-combobox-item-indicator">
-						<Check class="base-icon" />
-					</Combobox.ItemIndicator>
-					<div class="field-combobox-item-label">
-						{option.label}
+			{#each filteredOptions as group}
+				{#if 'options' in group}
+					<div class="field-combobox-group">
+						<div class="field-combobox-group-label">{group.label}</div>
+						{#each group.options as option}
+							<Combobox.Item value={option.value} label={option.label} class="field-combobox-item">
+								<Combobox.ItemIndicator class="field-combobox-item-indicator">
+									<Check class="base-icon" />
+								</Combobox.ItemIndicator>
+								<div class="field-combobox-item-label">
+									{option.label}
+									{#if option.extra}
+										<Badge>{option.extra}</Badge>
+									{/if}
+								</div>
+							</Combobox.Item>
+						{/each}
 					</div>
-				</Combobox.Item>
+				{:else}
+					<Combobox.Item value={group.value} label={group.label} class="field-combobox-item">
+						<Combobox.ItemIndicator class="field-combobox-item-indicator">
+							<Check class="base-icon" />
+						</Combobox.ItemIndicator>
+						<div class="field-combobox-item-label">
+							{group.label}
+							{#if group.extra}
+								<Badge>{group.extra}</Badge>
+							{/if}
+						</div>
+					</Combobox.Item>
+				{/if}
```

Comment:
@fmaclen maybe we could extract `<Combobox.Item>` to a separate component so we remove these duplicated lines, but I don't know if it's worth it.

Reply:
Yeah, we could DRY the `<Combobox.Item>` but I think your current version looks more readable. Let's keep it 👍